File: gateway/src/main/java/io/kuz/ecom/gateway/product/dto/ProductDTO.java
package io.kuz.ecom.gateway.product.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
public class ProductDTO {

    private int id;
    private String title;
    private int categoryId;
    private String categoryLabel;
    private List<ProductAttributeDTO> attributes;
    private BigDecimal price;
}

File: common/src/main/kotlin/io/kuz/ecom/common/dto/ProductDTO.kt
package io.kuz.ecom.common.dto

import java.math.BigDecimal

data class ProductDTO(
    val id: Int,
    val title: String,
    val categoryId: Int,
    val categoryLabel: String,
    val attributes: List<ProductAttributeDTO>,
    val price: BigDecimal,
)

data class ProductAttributeDTO(
    val id: Int,
    val label: String,
    val options: List<ProductAttributeOptionDTO>,
)

data class ProductAttributeOptionDTO(
    val id: Int,
    val attributeId: Int,
    val label: String,
)